plugins/labtests/models.py

```diff
@@ -0,0 +1,204 @@
+import datetime
+from django.utils.dateformat import format as dt_format
+from django.db import models
+from django.conf import settings
+from django.utils import timezone
+from opal.core import serialization
+from opal import models as omodels
+
+
+def format_dt(some_dt):
+    return dt_format(some_dt, settings.DATETIME_FORMAT)
+
+
+"""
+Models for labtests
+"""
+class LabTest(models.Model):
+    patient = models.ForeignKey(
+        omodels.Patient,
+        on_delete=models.CASCADE,
+        related_name="lab_tests"
+    )
+    clinical_info = models.TextField(blank=True)
+    datetime_ordered = models.DateTimeField(null=True, blank=True)
+    site = models.CharField(max_length=256, blank=True, null=True)
+    status = models.CharField(max_length=256, blank=True, null=True)
+    test_code = models.CharField(max_length=256, blank=True, null=True)
+    test_name = models.CharField(max_length=256, blank=True, null=True)
+    lab_number = models.CharField(max_length=256, blank=True, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def create_from_old_test(cls, lt):
+        s = cls()
+        s.patient = lt.patient
+        extras = lt.extras
+        fields = [
+            "clinical_info",
+            "site",
+            "status",
+            "test_code",
+            "test_name",
+            "lab_number",
+        ]
+        for f in fields:
+            setattr(s, f, extras.get(f, None))
```

Comment:
I'd swear a lot if this line ever turned up in a stacktrace :)

Reply:
resolved on 83b691d6